modules/storage-page-memory/src/test/java/org/apache/ignite/internal/storage/pagememory/index/InlineUtilsTest.java

```diff
@@ -0,0 +1,290 @@
+import static org.apache.ignite.internal.storage.pagememory.index.InlineUtils.inlineSize;
+import static org.apache.ignite.internal.storage.pagememory.index.InlineUtils.innerNodePayloadSize;
+import static org.apache.ignite.internal.storage.pagememory.index.InlineUtils.leafNodePayloadSize;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.EnumSet;
+import java.util.List;
+import java.util.stream.IntStream;
+import org.apache.ignite.internal.binarytuple.BinaryTupleCommon;
+import org.apache.ignite.internal.pagememory.tree.io.BplusInnerIo;
+import org.apache.ignite.internal.pagememory.tree.io.BplusLeafIo;
+import org.apache.ignite.internal.schema.NativeType;
+import org.apache.ignite.internal.schema.NativeTypeSpec;
+import org.apache.ignite.internal.schema.NativeTypes;
+import org.apache.ignite.internal.storage.index.IndexDescriptor;
+import org.apache.ignite.internal.storage.index.IndexDescriptor.ColumnDescriptor;
+import org.junit.jupiter.api.Test;
+
+/**
+ * For {@link InlineUtils} testing.
+ */
+public class InlineUtilsTest {
+    @Test
+    void testInlineSizeForNativeType() {
+        EnumSet<NativeTypeSpec> nativeTypeSpecs = EnumSet.allOf(NativeTypeSpec.class);
+
+        // Fixed length type checking.
+
+        NativeType nativeType = NativeTypes.INT8;
+
+        assertEquals(1, inlineSize(nativeType));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(2, inlineSize(nativeType = NativeTypes.INT16));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(4, inlineSize(nativeType = NativeTypes.INT32));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(8, inlineSize(nativeType = NativeTypes.INT64));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(4, inlineSize(nativeType = NativeTypes.FLOAT));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(8, inlineSize(nativeType = NativeTypes.DOUBLE));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(16, inlineSize(nativeType = NativeTypes.UUID));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(1, inlineSize(nativeType = NativeTypes.bitmaskOf(8)));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(3, inlineSize(nativeType = NativeTypes.DATE));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(4, inlineSize(nativeType = NativeTypes.time()));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(9, inlineSize(nativeType = NativeTypes.datetime()));
+        nativeTypeSpecs.remove(nativeType.spec());
+
+        assertEquals(12, inlineSize(nativeType = NativeTypes.timestamp()));
```

Comment:
12 bytes per timestamp? Jesus! Why is it so big? Just asking

Reply:
it's seconds (long) and nanos(int)